app/services/support_interface/notes_export.rb

```diff
@@ -0,0 +1,32 @@
+module SupportInterface
+  class NotesExport
+    LABELS = %w[
+      note_subject note_message note_created_at candidate_id provider_code provider_name provider_user_id
+      number_of_training_provider_relationships total_number_of_provider_relationships
+    ].freeze
+
+    def data_for_export
+      data = Note.joins(
+        [:provider_user, application_choice: [{ application_form: :candidate }, { course_option: { course: :provider } }]],
+      ).joins(
+        'LEFT JOIN provider_relationship_permissions tpp ON tpp.training_provider_id = providers.id',
+      ).joins(
+        'LEFT JOIN provider_relationship_permissions rpp ON rpp.ratifying_provider_id = providers.id',
+      ).group(
+        :subject, :message, :created_at, :"candidates.id", :"providers.code", :"providers.name", :"provider_users.id"
+      ).pluck(
+        :subject,
+        :message,
+        :created_at,
+        :"candidates.id",
+        :"providers.code",
+        :"providers.name",
+        :"provider_users.id",
+        'COUNT(tpp.id)',
+        Arel.sql('COUNT(tpp.id) + COUNT(rpp.id)'),
```

Comment:
These seem odd to include in a notes export. Can we check the need for this and make sure we're meeting it?

Reply:
Checked up about this and it is necessary for the analysis. I've since changed the query though as the way we derive the exported provider details and subsequent counts has changed.